comms/dht/tests/dht.rs

```diff
@@ -134,22 +142,23 @@ async fn setup_comms_dht(
         .build()
         .unwrap();
 
-    let dht = DhtBuilder::new(
+    let mut dht_builder = DhtBuilder::new(
         comms.node_identity(),
         comms.peer_manager(),
         outbound_tx,
         comms.connectivity(),
         comms.shutdown_signal(),
     )
     .local_test()
-    .disable_auto_store_and_forward_requests()
     .with_database_url(DbConnectionUrl::MemoryShared(random::string(8)))
     .with_discovery_timeout(Duration::from_secs(60))
-    .with_num_neighbouring_nodes(8)
-    .finish()
-    .await
-    .unwrap();
+    .with_num_neighbouring_nodes(8);
 
+    if with_saf {
```

Comment:
Did it break any tests in this suite to have it always enabled?

If not - I say that you can just always have it on and avoid having an extra boolean param

Reply:
It did break a few tests that made assumption on how many messages are sent and I was not comfortable updating them because I wasn't 100% sure what their intent was.

Reply:
Ah right, ye it makes sense to remove all the noise so that tests can remain focused

Reply:
Though now that I think about it - I think you should add an explicit call from node C to Node A (currently there is one for Node A -> C) - then I believe you wont need auto saf requests enabled

Reply:
Cool, done